src/libraries/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -32,18 +33,14 @@ internal static partial class Sys
         {
             const int StackBufferSize = 256;
 
-            // Use an initial buffer size that prevents disposing and renting
-            // a second time when calling ConvertAndTerminateString.
-            using var converter = new ValueUtf8Converter(stackalloc byte[StackBufferSize]);
             Span<byte> spanBuffer = stackalloc byte[StackBufferSize];
             byte[]? arrayBuffer = null;
-            ref byte pathReference = ref MemoryMarshal.GetReference(converter.ConvertAndTerminateString(path));
             while (true)
             {
                 int error = 0;
                 try
                 {
-                    int resultLength = ReadLink(ref pathReference, ref MemoryMarshal.GetReference(spanBuffer), spanBuffer.Length);
+                    int resultLength = ReadLink(path, ref MemoryMarshal.GetReference(spanBuffer), spanBuffer.Length);
```

Comment:
This change is introducing minor inefficiency in that the path argument will be converted to UTF8 more than when the code reloops for links longer than 256 characters. It should be super rare. I do not think it is worth it to complicate the code for it.